### PR TITLE
get commit blob with test case

### PR DESCRIPTION
**Pull request by ad-47**

### CODE REVIEW

**Comment by ravishetye** [comment]
@conradoverta : this LGTM, to be  merged to the @evgeniilezhnin's feature branch

**Comment by ravishetye** [comment]
this will change and possibly simplified by combining the two calls https://github.com/VertaAI/modeldb/pull/172

**Comment by ravishetye** [comment]
so the problem reduces to given a repo and commit  we traverse the location and return the folder or tree at that location. so picking up from your DM if the location provided is `[ 'deployment','train','backend','protos'] ` then in the commit we go to deployment/train/backend/protos and return a BLOB if protos is a BLOB and return tree if protos is a TREE

**Comment by evgeniilezhnin** [comment]
So I tried to test it before "combine protos" change and this endpoint returns all blob components combined from location[0].
Expected behaviour is to return a blob from specified location, right?

**Comment by evgeniilezhnin** [comment]
Another thing that in a base I added root folder sha to commit in database, so now commit hash and root folder hash is not the same.
Please use this diff to fix it: https://github.com/VertaAI/modeldb/commit/4505ded90c1b85bd535d06d9830de63890c2af71

File: backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java [hunk 1]
```diff
@@ -253,4 +255,154 @@ private String computeSHA(PathDatasetComponentBlob path) throws NoSuchAlgorithmE
...
+            () -> new ModelDBException("No such folder found", Status.Code.NOT_FOUND));
+  }
+
+  @Override
+  public GetCommitComponentRequest.Response getCommitComponent(
+      RepositoryFunction repositoryFunction, String commitHash, ProtocolStringList locationList)
+      throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      repositoryFunction.apply(session);
```
[the remark this answers is not in the document]
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]: added some validation

**Pull request merged**